### PR TITLE
update GLCI (warnings + docs)

### PR DESCRIPTION
**Pull request by ben-schwen**
- [x] Removes known false positives in CI
- [x] Updates .CI readme

[CI results](https://gitlab.com/Rdatatable/data.table/-/pipelines/1504133843)

### CODE REVIEW

File: .ci/README.md [hunk 1]
```diff
@@ -44,3 +44,22 @@ Base R implemented helper script, [originally proposed to base R](https://svn.r-
 ### [`publish.R`](./publish.R)
 
 Base R implemented helper script to orchestrate generation of most artifacts and to arrange them nicely. It is being used only in [_integration_ stage in GitLab CI pipeline](./../.gitlab-ci.yml).
+
+## GitLab Open Source Program
+
+We are currently part of the GitLab for [Open Source Program](https://about.gitlab.com/solutions/open-source/). This gives us 50,000 compute minutes per month for our GitLab CI. Our license needs to renewed yearly and is currently managed by @ben-schwen.
```
Comment (Anirban166) [inline .ci/README.md:50]: needs to be renewed

File: .ci/README.md [hunk 2]
```diff
@@ -44,3 +44,22 @@ Base R implemented helper script, [originally proposed to base R](https://svn.r-
...
+
+## Updating CI pipeline
+
+Basic CI checks are also run on every push to the GitLab repository. This can **and should** be used for PRs changing the CI pipeline before merging them to master.
+
+```shell
+# fetch changes from remote (GitHub) and push them to GitLab 
+git fetch git@github.com:Rdatatable/data.table.git new_branch:new_branch
+git push
+# after updating at GitHub, pull changes from remote and push to GitLab
```
Comment (Anirban166) [inline .ci/README.md:60]: updating on GitHub

File: .ci/README.md (hunk 1 above)
Comment (Anirban166) [reply .ci/README.md:50]: Also maybe extend the link, like \[GitLab for Open Source Program](https://about.gitlab.com/solutions/open-source/)

File: .ci/README.md [hunk 3]
```diff
@@ -44,3 +44,22 @@ Base R implemented helper script, [originally proposed to base R](https://svn.r-
...
 
 Base R implemented helper script to orchestrate generation of most artifacts and to arrange them nicely. It is being used only in [_integration_ stage in GitLab CI pipeline](./../.gitlab-ci.yml).
+
+## GitLab Open Source Program
+
+We are currently part of the GitLab for [Open Source Program](https://about.gitlab.com/solutions/open-source/). This gives us 50,000 compute minutes per month for our GitLab CI. Our license needs to renewed yearly and is currently managed by @ben-schwen.
+
+## Updating CI pipeline
+
+Basic CI checks are also run on every push to the GitLab repository. This can **and should** be used for PRs changing the CI pipeline before merging them to master.
```
Comment (jangorecki) [inline .ci/README.md:54]: Very good addition

**Review by jangorecki — APPROVED** [review]
LGTM

File: .ci/README.md (hunk 2 above)
Comment (ben-schwen) [reply .ci/README.md]: TY!

File: .ci/README.md (hunk 1 above)
Comment (ben-schwen) [reply .ci/README.md]: TY!

File: .ci/README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # data.table continuous integration and deployment
 
-On each Pull Request opened in GitHub we run GitHub Actions test jobs to provide prompt feedback about the status of PR. Our main CI pipeline runs on GitLab CI nightly. GitLab repository automatically mirrors our GitHub repository and runs pipeline on `master` branch every night. It tests more environments and different configurations. It publish variety of artifacts.
+On each Pull Request opened in GitHub we run GitHub Actions test jobs to provide prompt feedback about the status of PR. Our main CI pipeline runs on GitLab CI nightly. GitLab repository automatically mirrors our GitHub repository and runs pipeline on `master` branch every night. It tests more environments and different configurations. It publish variety of artifacts such as our [homepage](https://rdatatable.gitlab.io/data.table/) and [CRAN-like website for dev version](https://rdatatable.gitlab.io/data.table/web/packages/data.table/index.html) including windows binaries for the dev version.
```
Comment (MichaelChirico) [inline .ci/README.md:3]:
```suggestion
On each Pull Request opened in GitHub we run GitHub Actions test jobs to provide prompt feedback about the status of PR. Our more thorough main CI pipeline runs nightly on GitLab CI. GitLab repository automatically mirrors our GitHub repository and runs pipeline on `master` branch every night. It tests more environments and different configurations. It publishes a variety of artifacts such as our [homepage](https://rdatatable.gitlab.io/data.table/) and [CRAN-like website for dev version](https://rdatatable.gitlab.io/data.table/web/packages/data.table/index.html), including windows binaries for the dev version.
```

File: .ci/README.md [hunk 5]
```diff
@@ -44,3 +44,22 @@ Base R implemented helper script, [originally proposed to base R](https://svn.r-
 ### [`publish.R`](./publish.R)
 
 Base R implemented helper script to orchestrate generation of most artifacts and to arrange them nicely. It is being used only in [_integration_ stage in GitLab CI pipeline](./../.gitlab-ci.yml).
+
+## GitLab Open Source Program
+
+We are currently part of the [GitLab for Open Source Program](https://about.gitlab.com/solutions/open-source/). This gives us 50,000 compute minutes per month for our GitLab CI. Our license needs to be renewed yearly and is currently managed by @ben-schwen.
```
Comment (MichaelChirico) [inline .ci/README.md:50]: > yearly

Maybe give a rough idea of what time of year?

File: .ci/README.md [hunk 6]
```diff
@@ -44,3 +44,22 @@ Base R implemented helper script, [originally proposed to base R](https://svn.r-
...
+```shell
+# fetch changes from remote (GitHub) and push them to GitLab 
+git fetch git@github.com:Rdatatable/data.table.git new_branch:new_branch
+git push
+# after updating on GitHub, pull changes from remote and push to GitLab
+git pull git@github.com:Rdatatable/data.table.git new_branch
+git push
+```
+
+Make sure to include a link to the pipeline results in your PR.
```
Comment (MichaelChirico) [inline .ci/README.md:65]: Might be nice to have the GLCI pipeline automatically post a status to GitHub _for PRs touching GLCI files_. If you think that's feasible/worth it, feel free to mark as a follow-up issue.

Comment (ben-schwen) [reply .ci/README.md:65]: I guess the easy way to do this, is to make a GitHub action which is triggered on changes to `.gitlab-ci.yml`. Afterwards, it pushes to GitLab using a PAT and also posts the Pipeline link in the PR.

**Pull request merged**